nemo/utils/app_state.py

```diff
@@ -47,6 +47,17 @@ def __init__(self, device=None):
         self._module_registry = ObjectRegistry("module")
         # Create graph manager (registry with some additional functionality).
         self._neural_graph_manager = NeuralGraphManager()
+        # Create NmTensor registry
+        self._nmtensor_name_registry = nemo.core.neural_types.NmTensorNameRegistry()
+
+    @property
+    def tensor_names(self):
```

Comment:
... And in here you suggest that this is only mapping from one name to the other.

So what is really TensorRegistry storing?

Reply:
Yep NmTensorRegistry is just a mapping of user's naming of nmtensors to their unique_names. It also keeps track of all unique_names so we can initialize the TrainingState object.